pkg/oc/cli/admin/release/info.go

```diff
@@ -71,7 +74,32 @@ type InfoOptions struct {
 	Verify       bool
 }
 
-func (o *InfoOptions) Complete(cmd *cobra.Command, args []string) error {
+func (o *InfoOptions) Complete(f kcmdutil.Factory, cmd *cobra.Command, args []string) error {
+	if len(args) == 0 {
+		cfg, err := f.ToRESTConfig()
+		if err != nil {
+			return fmt.Errorf("info expects one argument, or a connection to a 4.0 OpenShift server: %v", err)
+		}
+		client, err := configv1client.NewForConfig(cfg)
+		if err != nil {
+			return fmt.Errorf("info expects one argument, or a connection to a 4.0 OpenShift server: %v", err)
+		}
+		cv, err := client.Config().ClusterVersions().Get("version", metav1.GetOptions{})
+		if err != nil {
+			if errors.IsNotFound(err) {
+				return fmt.Errorf("you must be connected to a v4.0 OpenShift server to fetch the current version")
+			}
+			return fmt.Errorf("info expects one argument, or a connection to a 4.0 OpenShift server: %v", err)
+		}
+		image := cv.Status.Current.Payload
+		if len(image) == 0 && cv.Spec.DesiredUpdate != nil {
+			image = cv.Spec.DesiredUpdate.Payload
+		}
+		if len(image) == 0 {
+			return fmt.Errorf("the server is not reporting a release image at this time, please specify an image to view")
+		}
+		args = []string{image}
+	}
 	if len(args) < 1 {
```

Comment:
I prefer when we fail fast, I still remember you've been picky about it in one of my early PRs.